### PR TITLE
json-Schema changes to comply with deep/shallow structure

### PR DESCRIPTION
**Pull request by Abdiiir**

### CODE REVIEW

**Comment by Abdiiir** [comment]
Note that, the TNT/API test is expected to fail as the schemas have requirements pertaining to the new API specification. 

- Can be reviewed but should NOT be merged.

File: src/test/resources/tnt/v2/EventSchema.json [hunk 1]
```diff
@@ -112,13 +112,13 @@
...
           "id": "#/transportCall/properties/otherFacility",
-          "type": "string",
+          "type":  ["null","string"],
           "default": "",
           "maxLength": 6
         },
         "modeOfTransport": {
           "id": "#/transportCall/properties/modeOfTransport",
-          "enum":  [ "VESSEL", "RAIL", "TRUCK", "BARGE" ]
+          "type": ["null","string"]
```
Comment (HenrikHL) [inline src/test/resources/tnt/v2/EventSchema.json]: Don't we have the enum type?
...and null should not be allowed - it is a mandatory field...

Comment (Abdiiir) [reply src/test/resources/tnt/v2/EventSchema.json]: I am aware, but a null value is still received in the test data. I wanted to add a comment but cannot do in JSON file. 
I was thinking that we can add it as a bug? 
PS: many test cases fail if this specification is upheld

Comment (HenrikHL) [reply src/test/resources/tnt/v2/EventSchema.json]: Not quite sure what 

> ...but a null value is still received in the test data...

means... I hope you are creating test cases based on the API specs and not the test data :-D

Comment (Abdiiir) [reply src/test/resources/tnt/v2/EventSchema.json]: I do, I just diverted in this specific case. 
I can revert it and let the tests fail? or leave as is and create a bug issue for it? 

testdata includes ->> `modeOftransport: null`

**Review by HenrikHL — APPROVED** [review]

**Pull request merged**